spinalcordtoolbox/gui/widgets.py

```diff
@@ -280,23 +321,36 @@ def refresh(self):
         self._x, self._y, self._z = [int(i) for i in self._parent._controller.position]
         data = self._image.data[self._x, :, :]
         self.view.set_array(data)
+        self.plot_position()
+        self.plot_points()
         self.view.figure.canvas.draw()
 
-    def on_update(self, event):
+    def on_select_point(self, event):
         if event.xdata > 0 and event.ydata > 0 and event.button == 1:
             self.point_selected_signal.emit(self._x, event.ydata, event.xdata)
 
     def plot_points(self):
-        logger.debug('Plotting points {}'.format(self._parent._controller.points))
-        if self._parent._controller.points:
-            points = [x for x in self._parent._controller.points]
+        if self._plot_points:
+            controller = self._parent._controller
+            logger.debug('Plotting points {}'.format(controller.points))
+            points = [x for x in controller.points if x[0] == controller.position[0]]
```

Comment:
I think it's better to add some comments on what is happening here, why the `except ValueError` below is required, etc.